presto-raptor/src/main/java/com/facebook/presto/raptor/metadata/DatabaseShardManager.java

```diff
@@ -365,6 +366,12 @@ private void runCommit(long transactionId, HandleConsumer callback)
                 return;
             }
             catch (DBIException e) {
+                if (isTransactionCacheFullError(e)) {
+                    throw metadataError(e,
+                            "A possible cause of this problem is the transaction size is too big " +
```

Comment:
How about this:
"A possible cause can be inserting too many shards into a table that has many columns."